sdk/python/tests/unit/permissions/auth/server/mock_utils.py

```diff
@@ -68,8 +68,7 @@ def _mock_kubernetes(request, monkeypatch):
         "feast.permissions.auth.kubernetes_token_parser.client.RbacAuthorizationV1Api.list_cluster_role_binding",
         lambda *args, **kwargs: clusterrolebindings["items"],
     )
-    m = mock_open(read_data="my-token")
     monkeypatch.setattr(
-        "builtins.open",
-        m,
+        "feast.permissions.client.kubernetes_auth_client_manager.KubernetesAuthClientManager.get_token",
```

Comment:
this way we're entirely mocking our business code, not the external API, so we're not covering the `get_token` function. WDYT?

Reply:
I agree but the way it was written before its problematic because it affect all open() calls in the code not only for the code inside our get_token()